### PR TITLE
refactored firetrader.js to support tce and multiple eq traders

### PR DESCRIPTION
**Pull request by Kastang**
See Discord for additional details.

### CODE REVIEW

File: commands/firetrade.js [hunk 1]
```diff
@@ -1,109 +1,162 @@
...
+      }
+    });
   }
 };
 
 exports.conf = {
   enabled: true,
   guildOnly: false,
-  aliases: ['ft', 'fire'],
+  aliases: ['tr', 'trade'],
```
Comment (TheCodeSinger) [inline commands/firetrade.js]: I don't like the mismatch of the filename now. I'd prefer to either 1. create a new command file called trade.js and delete firetrade.js or rename this one to trade.js.

I don't think the filename always has to match the primary command name, but in this case it's quite a deviation.

File: commands/firetrade.js [hunk 2]
```diff
@@ -1,109 +1,162 @@
...
-          'url': 'https://arsonwarehouse.com/!/lordoffire/bids/' + id + '@2x.png?timestamp=' + timestamp
-        },
-        'footer': {
-          'text': 'Fire Trading'
-        }
-      }
-    };
-  }
+  const item_list_cache = client.tornData.items;
+  const traders = client.config.params.firetrade.traders;
```
Comment (TheCodeSinger) [inline commands/firetrade.js]: is the `firetrade` prop name a holdover to the old naming? I would think it should just be `client.config.params.traders`

File: commands/firetrade.js [hunk 3]
```diff
@@ -1,109 +1,162 @@
...
+    found_match = true;
+    let matches_string = '';
+    partial_matches.forEach(match => {
+      matches_string += `\`!tr ${match}\`\n`
+    });
+
+    return message.channel.send({
+      embed: {
+        description: `
+          Partial matches found for "${item_name}, use one of the below commands:
```
Comment (TheCodeSinger) [inline commands/firetrade.js]: fully enclose the var with quotes. You left out the closing quote.   "${item_name}"
Same in line 118, above.

File: commands/firetrade.js [hunk 4]
```diff
@@ -1,109 +1,162 @@
...
-      return message.reply('More than 9 items found. Be more precise, please!');
+  // !tr <Item Name>
+  let found_match = false;
+  let partial_matches = [];
+  let item_name = args.join(' ');
+  Object.keys(item_list_cache).some(function(item_id, index) {
+    let normalized_item_name = item_list_cache[item_id]['name']
+
+    // if an exact match is found
+    // e.g. `!tr xanex`
```
Comment (TheCodeSinger) [inline commands/firetrade.js]: lol, your example exact match is wrong. It's "Xanax".

**Review by TheCodeSinger — CHANGES_REQUESTED** [review]

**Comment by owski** [comment]
Thank you for the suggestions. They were incorporated and `firetrade.js` was renamed to `trade.js`.

Will validate on stage shortly

**Comment by Kastang** [comment]
Thank you for the suggestions, the latest commit should account for them all.

I will be pushing to stage for a short test before going to production.

**Review by TheCodeSinger — APPROVED** [review]

**Review by TheCodeSinger — APPROVED** [review]

**Review by TheCodeSinger — APPROVED** [review]

**Pull request merged**